lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,491 @@
+   * Gets info about all non-settled channels.
+   * @param token_address an optional parameter to specify channels belonging to the specified token network
+   */
+  public getChannels = async (token_address?: string): Promise<Channel[]> => {
+    return ({} as any);
+  }
+
+  public channelBalance = async (
+    currency?: string,
+  ): Promise<ChannelBalance> => {
+    const client = this.getConnextClient();
+    if (!currency) {
+      return { balance: 0, pendingOpenBalance: 0, inactiveBalance: 0 };
+    }
+
+    const tokenAddress = this.getTokenAddress(currency);
+
+    const freeBalance = await client.getFreeBalance(tokenAddress);
+
+    const pendingOpenBalance = this.unitConverter.unitsToAmount({
+      currency,
+      units: freeBalance[client.multisigAddress].toNumber(),
+    });
+
+    const inactiveBalance = 0;
+
+    const balance = pendingOpenBalance + inactiveBalance;
+
+    return {
+      balance,
+      pendingOpenBalance,
+      inactiveBalance,
+    };
+  }
+
+  public tradingLimits = async (): Promise<TradingLimits> => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return {
+      maxSell: MAX_AMOUNT,
+      maxBuy: MAX_AMOUNT,
+    };
+  }
+
+  /**
+   * Returns the balances available in wallet for a specified currency.
+   */
+  public walletBalance = async (_currency?: string): Promise<WalletBalance> => {
+    return { totalBalance: 0, confirmedBalance: 0, unconfirmedBalance: 0 };
+  }
+  /**
+   * Creates a payment client.
+   */
+  public openChannel = async (): Promise<void> => {
+    if (!this.client) {
+      await this.initConnextClient();
+    }
+  }
+
+  /**
+   * Closes a payment client.
+   * @param channel_address the address of the client to close
+   */
+  public closeChannel = async (): Promise<void> => {
+    // not relevant for connext
+  }
+
+  /**
+   * Sends a token payment through the Connext network.
+   * @param target_address recipient of the payment
+   * @param token_address contract address of the token
+   * @param amount
+   * @param secret_hash optional; provide your own secret hash
+   */
+  private executePaymentTransfer = async (
+    payload: TokenPaymentRequest,
+  ): Promise<string> => {
+    const client = this.getConnextClient();
+    const transfer = await client.transfer({
+      recipient: payload.target_address,
+      amount: `${payload.amount}`,
+      meta: { secret_hash: payload.secret_hash },
+      assetId: payload.token_address,
+    });
+
+    const response: TokenPaymentResponse = {
+      ...payload,
+      secret: transfer.paymentId,
+    };
+
+    if (response.secret) {
+      // remove '0x'
+      return response.secret.slice(2);
+    } else {
+      throw errors.INVALID_TOKEN_PAYMENT_RESPONSE;
+    }
+  }
+
+  /**
+   * Deposits more of a token to an existing client.
+   * @param channel_address the address of the client to deposit to
+   * @param balance the amount to deposit to the client
+   */
+  public depositToChannel = async (
```

Comment:
If there are no plans to use this method or no corresponding call within Connext, I think we can just remove it.